### PR TITLE
MODFEE-104 create cancel endpoint, add tests

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Approach
- create cancel endpoint;
- add validation;
- create tests;

Resolves: [MODFEE-82](https://issues.folio.org/browse/MODFEE-82)

### CODE REVIEW

File: descriptors/ModuleDescriptor-template.json [hunk 1]
```diff
@@ -275,6 +275,19 @@
             "modperms.feesfines.patron-notices.post"
           ]
         },
+        {
+          "methods":[
+            "POST"
```
Comment (alexanderkurash) [inline descriptors/ModuleDescriptor-template.json:280]: Increase minor interface version

File: ramls/accounts.raml [hunk 2]
```diff
@@ -14,7 +14,8 @@ types:
   patronNotice: !include patronNotice.json
   checkActionRequest: !include actions/checkActionRequest.json
   checkActionResponse: !include actions/checkActionResponse.json
-  actionRequest: !include actions/actionRequest.json
+  defaultActionRequest: !include actions/actionRequest.json
```
Comment (alexanderkurash) [inline ramls/accounts.raml:17]: actionRequest.json -> defaultActionRequest.json

File: src/main/java/org/folio/rest/domain/ActionRequest.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package org.folio.rest.domain;
+
+public interface ActionRequest {
+
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/domain/ActionRequest.java:4]: Remove blank lines

File: src/main/java/org/folio/rest/impl/AccountsAPI.java [hunk 4]
```diff
@@ -459,8 +477,11 @@ else if (asyncResult.failed()) {
       if (cause instanceof FailedValidationException) {
         ActionFailureResponse response = new ActionFailureResponse()
           .withAccountId(accountId)
-          .withAmount(request.getAmount())
           .withErrorMessage(errorMessage);
+        if (request instanceof DefaultActionRequest) {
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/impl/AccountsAPI.java:481]: Check for CANCEL action type explicitly

File: src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  public Future<ActionValidationResult> validate(Account account, String rawAmount) {
+
+    validateIfAccountExists(account);
+
+    if (isClosed(account)) {
+      throw new FailedValidationException("Account is already closed");
+    }
+
+    return succeededFuture(new ActionValidationResult(
+      "0.00", null)
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java:39]: Use MonetoryValue

File: src/test/java/org/folio/rest/utils/ResourceClients.java [hunk 6]
```diff
@@ -36,6 +36,10 @@ public static ResourceClient accountsTransferClient(String accountId) {
     return buildUrlForAction(accountId, "transfer");
   }
 
+  public static ResourceClient accountsCancelClient(String accountId) {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/utils/ResourceClients.java:39]: Let's rename this method to buildAccountCancelClient  (also, rename others similarly)

File: src/test/java/org/folio/rest/utils/ResourceClients.java [hunk 7]
```diff
@@ -36,6 +36,10 @@ public static ResourceClient accountsTransferClient(String accountId) {
...
+    return buildUrlForAction(accountId, "cancel");
+  }
+
   private static ResourceClient buildUrlForAction(String accountId, String action) {
```
Comment (alexanderkurash) [inline src/test/java/org/folio/rest/utils/ResourceClients.java:43]: buildUrlForAction -> buildAccountActionClient

File: descriptors/ModuleDescriptor-template.json (hunk 1 above)
Comment (roman-barannyk) [reply descriptors/ModuleDescriptor-template.json:280]: done

File: ramls/accounts.raml (hunk 2 above)
Comment (roman-barannyk) [reply ramls/accounts.raml]: renamed

File: src/main/java/org/folio/rest/domain/ActionRequest.java (hunk 3 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/domain/ActionRequest.java]: removed

File: src/main/java/org/folio/rest/impl/AccountsAPI.java (hunk 4 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/impl/AccountsAPI.java]: updated

File: src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java (hunk 5 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java]: done

File: src/test/java/org/folio/rest/utils/ResourceClients.java (hunk 6 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/utils/ResourceClients.java]: renamed

File: src/test/java/org/folio/rest/utils/ResourceClients.java (hunk 7 above)
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/utils/ResourceClients.java]: renamed

File: src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+  @Override
+  public Future<ActionValidationResult> validate(Account account, String rawAmount) {
+
+    validateIfAccountExists(account);
+
+    if (isClosed(account)) {
+      throw new FailedValidationException("Account is already closed");
+    }
+
+    MonetaryValue remainingAmount = new MonetaryValue("0.00");
```
Comment (alexanderkurash) [inline src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java:50]: BigDecimal.ZERO ?

**Review by alexanderkurash — APPROVED** [review]

File: src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java (hunk 8 above)
Comment (roman-barannyk) [reply src/main/java/org/folio/rest/service/action/validation/CancelActionValidationService.java]: done

**Pull request merged**